### ISSUE TITLE
Own Loss Function in KERAS

### ISSUE THREAD

**Issue by AyanKumarBhunia** [fchollet/keras#7356]
1. How can I define my own loss function which required Weight and Bias parameters from previous layers in Keras?

2. How can I get [W1, b1, W2, b2, Wout, bout] from every layer? Here, we need to pass few more variable than usual (y_true, y_pred). I have attached two images for your reference.

I need to implement this loss function.

W(H) denote the linear prediction functions of the class labels based on the outputs of the latent layer. W(l) is the weight matrix of the l-th layer. To formalize the quantization error, we use H and B to represent the outputs of the latent layer and the expected binary matrix. The ideal binary matrix can be computed by B = sign(H). Beta is a non-negative parameter.

![capture](https://user-images.githubusercontent.com/9040741/28249837-2c50e3da-6a7b-11e7-9ecb-12d77f5d3eed.PNG)
![capture2](https://user-images.githubusercontent.com/9040741/28249836-2c50e100-6a7b-11e7-95a3-5abdd73e381d.PNG)

Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [ Yes] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [ Yes] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [Yes ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ Yes] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by alyato** [comment]
hi @AyanKumarBhunia  Do you solve it and share your own loss function?
And i also curious about this network.
![qq20180503-195613](https://user-images.githubusercontent.com/6815071/39575158-2084bcf4-4f0c-11e8-997d-cfd887772523.png)
 I think i can use the multi-outputs about above picture.
If you share how to implement the network ,it will cool.
Thanks.
